dlt/extract/incremental/__init__.py

```diff
@@ -371,13 +312,36 @@ def bind(self, pipe: SupportsPipe) -> "Incremental[TCursorValue]":
         logger.info(f"Bind incremental on {self.resource_name} with initial_value: {self.initial_value}, start_value: {self.start_value}, end_value: {self.end_value}")
         # cache state
         self._cached_state = self.get_state()
+        self._make_transformers()
         return self
 
     def __str__(self) -> str:
         return f"Incremental at {id(self)} for resource {self.resource_name} with cursor path: {self.cursor_path} initial {self.initial_value} lv_func {self.last_value_func}"
 
+    def _get_transformer(self, items: TDataItems) -> IncrementalTransformer:
+        # Assume list is all of the same type
+        for item in items if isinstance(items, list) else [items]:
+            if is_arrow_item(item):
+                return self._transformers['arrow']
+            elif pd is not None and isinstance(item, pd.DataFrame):
+                return self._transformers['arrow']
+            return self._transformers['json']
+        return self._transformers['json']
+
+    def __call__(self, rows: TDataItems, meta: Any = None) -> Optional[TDataItems]:
+        if rows is None:
+            return rows
+
+        transformer = self._get_transformer(rows)
+        # TODO: Primary key is not applied at bind time?
```

Comment:
yes - in case incremental is on transformer. then we call bind before _incremental in wrapper is created (transformer is called for a first time already when executing in the pipe)

Reply:
Hmm ok, I see. Then it's only the primary key that might be deferred, right? I could also instatiate the transformers lazily in `__call__` if that's safer. What do you think?

Reply:
there's one test that uses incremental on transformer and it will fail (if there's any problem at all). also get the current devel. I improved this behavior (a little bit) recently